### DISCUSSION TITLE
error minting - invalid private key: invalid length, need 256 bits

### DISCUSSION THREAD

**Discussion by BrwGamer** [gochain/web3#222] [General]
This command:
`sudo web3 contract call --wait --abi TEST.abi --gas-limit 2000000 --function mint ox...`
throws this error:
`ERROR: Error calling contract: invalid private key: invalid length, need 256 bits`

Even if the variable WEB3_PRIVATE_KEY is set in the environment, I know this because the deploy reads that variable and it works.

**Comment by treeder** [comment]
Didn't you already ask this in #220 ?

**Reply by BrwGamer** [reply]
no, this error happens for a different command. As I said, here it don't find the private key even if it is set.

**Reply by BrwGamer** [reply]
And if I run it without `sudo`:
`web3 contract call --wait --abi TEST.abi --gas-limit 2000000 --function mint ox...`
this is the error:
`ERROR: error getting ABI: Get "TEST.abi": unsupported protocol scheme ""`

**Reply by BrwGamer** [reply]
Even if I set all permissions to all users to the whole web3 project directory, when calling the `mint` function this error occurs:
`ERROR: Error calling contract: invalid private key: invalid length, need 256 bits`
This happens independently of the use of sudo.
The `WEB3_PRIVATE_KEY` is already stored with `export WEB3_PRIVATE_KEY=0x...` and inside `/etc/environment`. The same is for `WEB3_ADDRESS`.

Thanks

**Reply by treeder** [reply]
It's either not set or actually invalid. Be sure it's a correct private key.

**Reply by BrwGamer** [reply]
I'm pretty sure. I just used it to deploy the contract one more time and also to set the account on Metamask.
If I do `web3 env` it is set correctly.
Also by passing it to the mint command through `--pk` the error is the same.
What can I do to check?
Thanks

**Comment by NiSHoW** [comment]
@BrwGamer 
See this #223

**Comment by treeder** [comment]
This should be fixed in latest version.
